specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/assignmentOperation.json

```diff
@@ -0,0 +1,322 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          },
+          {
+            "$ref": "#/parameters/AssignmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AssignmentOperationNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Blueprint assignment retrieved.",
+            "schema": {
+              "$ref": "#/definitions/AssignmentOperation"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AssignmentOperation": {
+      "type": "object",
+      "description": "Represents underlying deployment detail for each update to the assignment.",
+      "allOf": [
+        { "$ref": "#/definitions/AzureResourceBase" }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Properties for AssignmentOperation",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AssignmentOperationProperties"
+        }
+      }
+    },
+    "AssignmentOperationList":{
+      "type": "object",
+      "description": "List of AssignmentOperation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of AssignmentOperation.",
+          "items": {
+            "$ref": "#/definitions/AssignmentOperation"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Link to the next page of results."
+        }
+      }
+
+    },
+    "AssignmentOperationProperties": {
+      "type": "object",
+      "description": "properties of AssignmentOperation.",
+      "properties": {
+        "blueprintVersion": {
+          "type": "string",
+          "description": "The blueprint version used for the assignment operation."
+        },
+        "assignmentState": {
+          "type": "string",
+          "description": "State of this assignment operation."
+        },
+        "timeCreated": {
+          "type": "string",
+          "description": "Create time of this Assignment Operation."
+        },
+        "timeStarted": {
+          "type": "string",
+          "description": "Start time of the underlying deployment."
+        },
+        "timeFinished": {
+          "type": "string",
+          "description": "Finish time of the overall underlying deployments."
+        },
+        "deployments": {
+          "type": "array",
+          "description": "list of jobs in this assignment operation.",
+          "items": {
+            "$ref": "#/definitions/AssignmentDeploymentJob"
+          }
+        }
+      }
+    },
+    "AssignmentDeploymentJob": {
+      "type": "object",
+      "description": "Represents individual job in given assignment operation.",
+      "properties": {
+        "kind": {
```

Comment:
Are any of these (kind or jobState for example) Enums? If so you should declare the possible values here:

Example:
        "intervalLength": {
          "type": "string",
          "description": "Interval value in minutes used to create LogAnalytics call rate logs.",
          "enum": [
            "ThreeMins",
            "FiveMins",
            "ThirtyMins",
            "SixtyMins"
          ],
          "x-ms-enum": {
            "name": "IntervalInMins",
            "modelAsString": false
          }
        }

Reply:
model as enum will force us to upgrade api-version on enum changes.

Will re-consider model them as enum when we are more stable and ready to GA.